### PR TITLE
New assignment4 @TylerRose8

### PR DESCRIPTION
**Pull request by schuywalker**
@TylerRose8

### CODE REVIEW

**Comment by TylerRose8** [comment]
Link:
https://gentle-forest-0aa0ed71e.1.azurestaticapps.net/game

File: Wordle.Api/Dtos/RecentStats.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+public class RecentStats
+{
+    private DateTime day { get; set; }
+    private double averageGuesses { get; set; }
+    private int averageTime { get; set; }
+    private double winRate { get; set; }
+    private bool wonDay { get; set; }
+    private int plays { get; set; }
+
+    public RecentStats(DateTime day, int plays, double averageGuesses, int averageTime, double winRate, bool wonDay)
```
Comment (tremorris1999) [inline Wordle.Api/Dtos/RecentStats.cs:11]: I see where this is going! I like it.

File: Wordle.Api/Data/Game.cs [hunk 2]
```diff
@@ -5,9 +5,10 @@ namespace Wordle.Api.Data;
 
 public class Game
 {
-    public enum GameTypeEnum{
+    public enum GameTypeEnum {
         Random = 1,
         WordOfTheDay = 2,
+        PlayedWordOfTheDay = 3,
```
Comment (tremorris1999) [inline Wordle.Api/Data/Game.cs:11]: This looks good!

File: Wordle.Api/Services/RecentStatsService.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    public class RecentStatsService
+    {
+        public RecentStatsService(AppDbContext context)
+        {
+            _context = context;
+        }
+
+        private readonly AppDbContext _context;
+
+        public IEnumerable<RecentStats> GetRecentScoreStats(string PlayerId)
```
Comment (tremorris1999) [inline Wordle.Api/Services/RecentStatsService.cs:15]: Nice! New service & controller seems a bit too labor intensive for my lazy style, but this definitely should work!

File: wordle-web/pages/game.vue [hunk 4]
```diff
@@ -70,108 +104,187 @@
...
+  }
+
+  picker = new Date(Date.now() - new Date().getTimezoneOffset() * 60000)
+    .toISOString()
+    .substr(0, 10)
+
+  wordleGame: WordleGame = new WordleGame(this.word)
+
+  pickDate() {
+    // This is gross
```
Comment (tremorris1999) [inline wordle-web/pages/game.vue:190]: Gross? I think it looks fine.

File: wordle-web/pages/instructions.vue [hunk 5]
```diff
@@ -0,0 +1,37 @@
+<template>
```
Comment (tremorris1999) [inline wordle-web/pages/instructions.vue:1]: This is good! You might want to add a section that shows what each color means on the game board.

File: wordle-web/pages/dailyLeaderboard.vue [hunk 6]
```diff
@@ -0,0 +1,53 @@
+<template>
```
Comment (tremorris1999) [inline wordle-web/pages/dailyLeaderboard.vue:1]: This page seems to display improperly when I run from your deployment, and when I run locally.

**Review by tremorris1999 — COMMENTED** [review]
## Features

- Make all pages responsive ✅ (These all look good to me!)
  - 1080p desktop ✅
  - iPad Air
  - Samsung Galaxy S20 Ultra Super Mega Excellent ✅
  - IPhone SE ✅
- Create a improved landing page and site theme (Background, make it look attractive to play) ✅
  - Theme should extend all pages (Maybe not the ad page 😁) ✅
- Add an instructions page that is available via the hamburger menu that details the features of the game and how to play ❌✅ (Needs more on "How to Play")
- Create a new Wordle game page that reuses components but will provide the same word for a specific day. Everyone going to the page that day will get the same word. The page should indicate Wordle of the Day or something like that ✅ (Same page, but still works)
- Create a page that lists the last 10 daily words with statistics. Number of plays, Average score, Average time. (just list the date, don't show the word silly) ✅
  - It should indicate if the user has played that day ✅
  - Clicking on the item should take the player to that game (Extra Credit) ❌

Overall it looks pretty good. I think you may have an issue with your deployment as I couldn't load the game page since the API request never gets resolved. I also couldn't run your API locally (platform incompatible with LocalDB), so I can't tell you if it's an issue with your API or with your front end. 

From what I can see, the pages look good in all those different sizes, so I suppose I'd call them responsive!

Great work guys! I can't wait to see this one finished and ironed out.

**Review by mmwoodfo — COMMENTED** [review]
Looks good guys. The PR should be going against main (I made that change) and you were technically supposed to make a new page for the daily word while keeping the old page, but the way you guys did it works just fine.

**Pull request closed without merging**